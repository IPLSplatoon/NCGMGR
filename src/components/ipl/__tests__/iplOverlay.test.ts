import IplOverlay from '@/components/ipl/iplOverlay.vue'
import { mount } from '@vue/test-utils'

describe('IplOverlay', () => {
    it('matches snapshot when visible', () => {
        const wrapper = mount(IplOverlay, {
            props: {
                visible: true
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when visible and max width', () => {
        const wrapper = mount(IplOverlay, {
            props: {
                visible: true,
                maxWidth: true
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when hidden', () => {
        const wrapper = mount(IplOverlay, {
            props: {
                visible: false
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('closes itself on background click if noBackgroundClose is false', () => {
        const wrapper = mount(IplOverlay, {
            props: {
                visible: true,
                noBackgroundClose: false
            }
        })

        wrapper.get('.ipl-overlay__wrapper').trigger('click')

        const emitted = wrapper.emitted('update:visible')
        expect(emitted?.length).toEqual(1)
        expect(emitted?.[0]).toEqual([false])
    })

    it('does nothing on background click if noBackgroundClose is true', () => {
        const wrapper = mount(IplOverlay, {
            props: {
                visible: true,
                noBackgroundClose: true
            }
        })

        wrapper.get('.ipl-overlay__wrapper').trigger('click')

        expect(wrapper.emitted('update:visible')).toBeFalsy()
    })
})
