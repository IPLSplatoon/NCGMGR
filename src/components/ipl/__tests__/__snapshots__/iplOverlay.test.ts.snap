// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplOverlay matches snapshot when hidden 1`] = `
<transition-stub>
  <!--v-if-->
</transition-stub>
`;

exports[`IplOverlay matches snapshot when visible 1`] = `
<transition-stub>
  <div class="ipl-overlay__wrapper layout horizontal center-horizontal center-vertical">
    <div class="ipl-overlay__content"></div>
  </div>
</transition-stub>
`;
